pylabrobot/resources/resource_stack.py

```diff
@@ -63,32 +63,50 @@ def __init__(
   def __str__(self) -> str:
     return f"ResourceGroup({self.name})"
 
-  def get_size_x(self) -> float:
+  @property
+  def _size_x(self) -> float:
     if len(self.children) == 0:
       return 0
     if self.direction == "x":
       return sum(child.get_size_x() for child in self.children)
     return max(resource.get_size_x() for resource in self.children)
 
-  def get_size_y(self) -> float:
+  @_size_x.setter
+  def _size_x(self, size_x: float):
+    raise AttributeError("Cannot set size_x for ResourceStack, use assign_child_resource instead")
+
+  @property
+  def _size_y(self) -> float:
     if len(self.children) == 0:
       return 0
     if self.direction == "y":
       return sum(child.get_size_y() for child in self.children)
     return max(resource.get_size_y() for resource in self.children)
 
-  def get_size_z(self) -> float:
+  @_size_y.setter
+  def _size_y(self, size_y: float):
+    raise AttributeError("Cannot set size_y for ResourceStack, use assign_child_resource instead")
+
+  @property
+  def _size_z(self) -> float:
```

Comment:
we had to change these since the previous `_size_z` didn't account for the addition of children